src/sysusers/sysusers.c

```diff
@@ -1099,10 +1099,17 @@ static int add_group(Item *i) {
                 r = gid_is_ok(i->gid);
                 if (r < 0)
                         return log_error_errno(r, "Failed to verify gid " GID_FMT ": %m", i->gid);
+                /* If we require the gid to already exist we can return here (either with an error or with nothing to do) */
+                if (i->gid_must_exist) {
+                        if (r > 0) {
```

Comment:
hmm, isn't this check reversed? not following here?

Reply:
Sorry, I'm not sure I understand. I improved the code comment but maybe I'm missing something. Happy to rework if you think this is confusing.

Reply:
ah, i was a bit confused. thanks for the longer comment, that helped me grok this.